src/Discord.Net.Rest/API/Common/Invite.cs

```diff
@@ -7,7 +7,7 @@ internal class Invite
         [JsonProperty("code")]
         public string Code { get; set; }
         [JsonProperty("guild")]
-        public Optional<InviteGuild> Guild { get; set; }
+        public Optional<Guild> Guild { get; set; }
```

Comment:
endpoint returns partial guild obj, shouldnt we expand InviteGuild to accommodate the missing few fields instead of deserializing it as a complete guild obj.

Reply:
I'm open for a discussion on this

Docs do not specifically mention which fields would be populated
So using a complete guild object might be better if they end up changing something

And also it allows to just populate existing `IGuild Guild` property of `RestInvite`

Reply:
~~Are we gonna make a new pr to add data each time they change something~~

Reply:
> Are we gonna make a new pr to add data each time they change something

Isn't that what a library is? An implementation of an API's current model..? If the API changes the library should change too...

Reply:
I agree with Cenngo here, Extra properties on objects that are never populated seems misleading

Reply:
> I think its best if we revert to InviteGuild here, can you make the necessary changes?

should be resolved now